app/src/main/java/com/chouten/app/ui/views/home/HomePageViewModel.kt

```diff
@@ -35,94 +37,89 @@ class HomePageViewModel(
-        isLoading = true
-        homeModule.subtypes.forEach { subtype ->
-            // wait for homeModule code to be loaded asynchronously using run handler
-            handler.postDelayed(object : Runnable {
-                override fun run() {
-                    println("Checking if homeModule code is loaded...")
-                    if (homeModule.code?.get(subtype)?.home?.get(0)?.code != null) {
-                        println("HomeModule code is loaded!")
-                        val homeFns = homeModule.code?.get(subtype)?.home
-                        if (homeFns != null) {
-                            viewModelScope.launch {
-                                runHomeFns(homeFns)
-                            }
-                        }
-
-                        handler.removeCallbacks(this)
-                        return
-                    }
-                    handler.postDelayed(this, 100)
-                }
-            }, 100)
+    private val handler = Handler(Looper.getMainLooper())
 
+    fun initialize(shouldReset: Boolean = false) {
+        if(shouldReset){
+            this.errors = 0;
         }
 
+        _homeResults.clear()
+        viewModelScope.launch { loadHomePage() }
     }
 
-    private suspend fun runHomeFns(homeFns: List<ModuleModel.ModuleCode.ModuleCodeblock>) {
-        homeFns.forEach { homeFn ->
-            println("HOME FN: $homeFn")
-            viewModelScope.launch {
-                if (!webview.load(
-                        homeFn,
-                    )
-                ) {
-                    isLoading = false
-                    return@launch
-                }
-
-                val res = webview.inject(homeFn)
-                if (res.isBlank()) {
-                    PrimaryDataLayer.enqueueSnackbar(
-                        SnackbarVisualsWithError(
-                            "Something went wrong while loading the home page.",
-                            false
-                        )
-                    )
-                    isLoading = false
-                    return@launch
-                }
-
-                try {
-                    val results =
-                        Mapper.parse<ModuleResponse<List<HomeResult>>>(
-                            res
-                        )
-                    println("RESULTS: $results")
-                    _homeResults.clear()
-                    _homeResults.addAll(results.result)
-                    isLoading = false
-                    webview.updateNextUrl(results.nextUrl)
-                } catch (e: Exception) {
-                    isLoading = false
-                    PrimaryDataLayer.enqueueSnackbar(
-                        SnackbarVisualsWithError(
-                            e.localizedMessage ?: "Error parsing home page results.", isError = true
-                        )
-                    )
-                    e.printStackTrace()
-                }
+    fun getCode(): String{
```

Comment:
Please view previous comments within the Info and Search viewmodels. This should be extracted into the ModuleDataLayer and can be simplified following advice from the Search VM.